harmony_netcdf_to_zarr/convert.py

```diff
@@ -99,7 +100,7 @@ def __copy_variable(src, dst_group, name):
         # Treat a 0-dimensional NetCDF variable as a zarr group
         dst = dst_group.create_group(name)
     else:
-        dtype = np.float64
+        dtype = src.dtype
```

Comment:
I don't know why this code is here, but it makes me wonder if `src.dtype` might not be set - should we have `src.dtype || np.float64`?

Reply:
Check with @rwat .  I remember he worked on this a bit initially and had some reasoning on this.

Reply:
My use of np.float64 here was intentional but Hailiang's decision is better. Some background documentation:

http://www.bic.mni.mcgill.ca/users/sean/Docs/netcdf/guide.txn_18.html

> When scale_factor and add_offset are used for packing, the associated variable (containing the packed data) is typically of type byte or short, whereas the unpacked values are intended to be of type float or double. The attributes scale_factor and add_offset should both be of the type intended for the unpacked data, e.g. float or double.

I picked np.float64 as the default to avoid any loss of precision but because the python netCDF4 library represents Variables in Datasets as numpy arrays (same as the python zarr library) it's guaranteed to have a dtype so using src.dtype to determine type is correct here. I shouldn't have written code that modifies the dtype when working with unscaled data (i.e. no scaling or offset specified) and Hailiang's code fixes my mistake.